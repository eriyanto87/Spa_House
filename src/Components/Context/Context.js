import React from "react";

const Context = React.createContext({
  // users: [],
  // cart: [],
  // treatments: [],
  // addUser: () => {},
  // addCart: () => {},
  //   getAllTreatments: () => {},
});

export default Context;
