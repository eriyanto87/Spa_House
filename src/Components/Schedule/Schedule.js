import React from "react";
import Context from "../Context/Context";
import moment from "moment";
import Nav from "../Nav/Nav";
import Footer from "../Footer/Footer";

function Schedule(props) {
  const goBack = (e, context) => {
    //set treatment back to 0 when the back button is pressed
    context.setTreatment(0);
    props.history.goBack();
  };

  const onSubmit = (evt, context) => {
    evt.preventDefault();

    const name = evt.target.name.value;
    const email = evt.target.email.value;
    const number = evt.target.number.value;

    const street = evt.target.street.value;
    const city = evt.target.city.value;
    const state = evt.target.state.value;
    //changed the zip code to integer
    const zip = Number(evt.target.zip.value);

    const date = evt.target.date.value;
    const time = evt.target.time.value;

    context.setDisplayDate(moment(date).format("LL"));
    context.setDisplayTime(time);

    const dt = new Date(`${date}T${time}`);
    const dts = dt.toISOString();

    //create a bad route(conflicting times or time isn't available)
    for (let i = 0; i < context.confirmations.length; i++) {
      const dtds = dt.toLocaleDateString();
      if (
        new Date(
          context.confirmations[i].appointment_date
        ).toLocaleDateString() == dtds
      ) {
        alert("Date already booked - Please select a different date");
        return;
      }
    }

    //cannot book prior to today
    if (dt <= new Date()) {
      alert("please select a later time");
    } else {
      const userData = {
        user_name: name,
        user_email: email,
        user_number: number,
        user_street: street,
        user_city: city,
        user_state: state,
        user_zip: zip,
      };
      context.setDatabaseDate(dts);
      context.addUser(userData);
      props.history.push("/cart");
    }
  };

  return (
    <Context.Consumer>
      {(context) => {
        return (
          <div>
            <Nav />
            <section>
              <h3>{context.abc}Book Your Appointment!</h3>
              <form
                className="datetimecontainer"
                onSubmit={(evt) => onSubmit(evt, context)}
              >
                <label>Contact Info</label>
                <p>
                  <input
                    required
                    placeholder="Name is required"
                    type="text"
                    name="name"
                  />
                </p>
                <p>
                  <input
                    required
                    placeholder="Phone number is required"
                    type="text"
                    name="number"
                  />
                </p>
                <p>
                  <input
                    placeholder="Enter your email address here"
                    type="text"
                    name="email"
                  />
                </p>
                <label>Address: </label>
                <p>
                  Please fill this out if you want us to come to your location
                </p>
                <p>
                  <input type="text" name="street" placeholder="Street" />
                </p>
                <p>
                  <input type="text" name="city" placeholder="City" />
                </p>
                <p>
                  <input type="text" name="state" placeholder="State" />
                </p>
                <p>
                  <input type="number" name="zip" placeholder="Zip" />
                </p>
                <p>
                  <label>Schedule a Date: </label>
                  <br />
                  <input required type="date" name="date" />
                </p>
                <p>
                  <label>Schedule a Time: </label>
                  <br />
                  <input required type="time" name="time" />
                </p>
                <button onClick={(e) => goBack(e, context)} type="cancel">
                  Back
                </button>{" "}
                <button id="cart" type="submit">
                  Cart
                </button>
              </form>
            </section>
            <Footer />
          </div>
        );
      }}
    </Context.Consumer>
  );
}

export default Schedule;
