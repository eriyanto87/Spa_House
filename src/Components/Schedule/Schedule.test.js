import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Context from "../Context/Context";
import Schedule from "./Schedule";

describe("Schedule component", () => {
  const contextValues = {
    setDisplayDate: () => {},
    setDisplayTime: () => {},
    setDatabaseDate: () => {},
    addUser: () => {},
  };

  it("Schedule Snapshot", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Context.Provider value={contextValues}>
            <Schedule />
          </Context.Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <Context.Provider value={contextValues}>
          <Schedule />
        </Context.Provider>
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
});
