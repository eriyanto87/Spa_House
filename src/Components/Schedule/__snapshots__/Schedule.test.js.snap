// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule component Schedule Snapshot 1`] = `
<div>
  <div
    className="nav"
  >
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home 
        </a>
      </li>
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          About 
        </a>
      </li>
      <li>
        <a
          href="/treatments"
          onClick={[Function]}
        >
          Treatments 
        </a>
      </li>
      <li>
        <a
          href="/pricing"
          onClick={[Function]}
        >
          Pricing 
        </a>
      </li>
      <li>
        <a
          href="/contact"
          onClick={[Function]}
        >
          Contact Us 
        </a>
      </li>
      <li>
        <a
          href="/gallery"
          onClick={[Function]}
        >
          Gallery
        </a>
      </li>
    </ul>
  </div>
  <section>
    <h3>
      Book Your Appointment!
    </h3>
    <form
      className="datetimecontainer"
      onSubmit={[Function]}
    >
      <label>
        Contact Info
      </label>
      <p>
        <input
          name="name"
          placeholder="Name is required"
          required={true}
          type="text"
        />
      </p>
      <p>
        <input
          name="number"
          placeholder="Phone number is required"
          required={true}
          type="text"
        />
      </p>
      <p>
        <input
          name="email"
          placeholder="Enter your email address here"
          type="text"
        />
      </p>
      <label>
        Address: 
      </label>
      <p>
        Please fill this out if you want us to come to your location
      </p>
      <p>
        <input
          name="street"
          placeholder="Street"
          type="text"
        />
      </p>
      <p>
        <input
          name="city"
          placeholder="City"
          type="text"
        />
      </p>
      <p>
        <input
          name="state"
          placeholder="State"
          type="text"
        />
      </p>
      <p>
        <input
          name="zip"
          placeholder="Zip"
          type="number"
        />
      </p>
      <p>
        <label>
          Schedule a Date: 
        </label>
        <br />
        <input
          name="date"
          required={true}
          type="date"
        />
      </p>
      <p>
        <label>
          Schedule a Time: 
        </label>
        <br />
        <input
          name="time"
          required={true}
          type="time"
        />
      </p>
      <button
        onClick={[Function]}
        type="cancel"
      >
        Back
      </button>
       
      <button
        id="cart"
        type="submit"
      >
        Cart
      </button>
    </form>
  </section>
  <div>
    <p>
      Â© 2020
    </p>
  </div>
</div>
`;
